src/main/frontend/handler/editor.cljs

```diff
@@ -801,10 +801,19 @@
                                              (seq (:block/_parent block)))))]
              (when-not (and has-children? left-has-children?)
                (when block-parent-id
-                 (let [block-parent (gdom/getElement block-parent-id)
-                       sibling-block (util/get-prev-block-non-collapsed-non-embed block-parent)]
-                   (delete-block-aux! block delete-children?)
-                   (move-to-prev-block repo sibling-block format id value)))))))))
+                 (cond
+                   (:embed? config)
+                   (let [block-parent (gdom/getElement block-parent-id)
+                         blocks-container (util/rec-get-blocks-container block-parent)
+                         sibling-block (util/get-prev-block-non-collapsed-in-embed blocks-container block-parent)]
```

Comment:
Looks like the only difference between these two branches is this fn. Looking at those two fns, the only difference I can see is blocks. I think we'd have less copy and paste code if we change `get-prev-block-non-collapsed-non-embed to take blocks

Reply:
Now I put the `cond` expression to the binding of `sibling-block`, making less repeated codes.